compiler/rustc_infer/src/infer/canonical/mod.rs

```diff
@@ -63,8 +63,8 @@ impl<'cx, 'tcx> InferCtxt<'cx, 'tcx> {
         // in them, so this code has no effect, but it is looking
         // forward to the day when we *do* want to carry universes
         // through into queries.
-        let universes: IndexVec<ty::UniverseIndex, _> = std::iter::once(ty::UniverseIndex::ROOT)
-            .chain((0..canonical.max_universe.as_u32()).map(|_| self.create_next_universe()))
+        let universes: IndexVec<ty::UniverseIndex, _> = std::iter::once(self.universe())
```

Comment:
I think we should update the comment above to read something like

"Instantiate the root-universe content into the current universe, and create fresh universes for other things."

It took me a bit to understand what this code did. It makes sense, though.

Reply:
```suggestion
        //
        // Instantiate the root-universe content into the current universe,
        // and create fresh universes for the higher universes.
        let universes: IndexVec<ty::UniverseIndex, _> = std::iter::once(self.universe())
```